### DISCUSSION TITLE
Upgrading from CSLA 3.8 to 5.5.3

### DISCUSSION THREAD

**Discussion by oskarjf** [MarimerLLC/csla#3271] [Questions] [answered]
Hi
I'm upgrading CSLA from 3.8 to 5.5 and I'm a bit stuck on one thing. 3.8 is using the ValidationRules.AddInstanceRule and I was wondering what the equal concept is in 5.5. Can't find anything when searching.

Also how would I do this in 5.5?
.AddRule(ObjectRules.PropertyIsUniqueRule<Role, string>, new ObjectRules.PropertyIsUniqueRuleArgs<string>(NameProperty, IdProperty, new ObjectRules.GetObjects<string>

Brg Oskar

**Comment by jonnybee** [comment, marked as answer]
CSLA 4 introduced a new Rule engine where rules are classes. 
So there is no longer any InstanceRules in CSLA 4 and higher. 

You will need to rewrite the logic using the new rule system.

**Reply by rockfordlhotka** [reply]
Though you (@jonnybee) did build the NuGet containing code to help ease the transition by wrapping the old-style rules. That package is still valid and available.

**Reply by oskarjf** [reply]
So I should just create a rule class inheriting from BusinesRule and implement my rule logic there?

**Reply by rockfordlhotka** [reply]
That is the best way to move forward if you can do so.

The "new" rules engine is incredibly powerful and can do far more than the old delegate model. There is a sample app in the Samples directory that demonstrates most of the rule engine capabilities.

**Comment by jonnybee** [comment]
I did write a blog post on Csla Validation some years ago: 
[CSLA Validation](https://jonnybekkum.wordpress.com/2013/10/07/csla-validationthe-new-nuget-package-in-csla-net-4-5-40/ ) and 
[CSLA 4.5 RuleEngine Update](https://jonnybekkum.wordpress.com/2012/11/07/csla-4-5-ruleengine-update/)

I would also recommend to look at the rules in [RuleTutorial Sample](https://github.com/MarimerLLC/csla/tree/main/Samples/RuleTutorial)
as they show different types of rules and hos they work.
